android/src/org/mozilla/firefox/vpn/qt/GleanUtil.kt

```diff
@@ -0,0 +1,121 @@
+package org.mozilla.firefox.vpn.qt
+
+import android.util.Log
+import mozilla.components.service.glean.BuildInfo
+import mozilla.components.service.glean.Glean
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.private.EventExtras
+import mozilla.telemetry.glean.net.HttpURLConnectionUploader
+import mozilla.telemetry.glean.private.EventMetricType
+import mozilla.telemetry.glean.private.NoExtraKeys
+import mozilla.telemetry.glean.private.NoExtras
+import org.json.JSONObject
+import org.mozilla.firefox.vpn.GleanMetrics.Pings
+import org.mozilla.firefox.vpn.GleanMetrics.Sample
+import java.lang.UnsupportedOperationException
+import java.util.*
+import kotlin.reflect.KProperty1
+
+// @MainThread
+@Suppress("UNUSED") // All the calls are via jni from the client, so we can ignore
+object GleanUtil {
+    @JvmStatic
+    fun initializeGlean(upload: Boolean = false, channel: String = "staging") {
+        val conf = Configuration(HttpURLConnectionUploader(), Configuration.DEFAULT_TELEMETRY_ENDPOINT, channel, null)
+        val build = BuildInfo("versionCode", "VersionName", Calendar.getInstance())
+        Glean.registerPings(Pings)
+        Glean.initialize(
+            applicationContext = VPNActivity.getInstance().applicationContext,
+            uploadEnabled = upload,
+            buildInfo = build,
+            configuration = conf
+        )
+        Log.i("Android-Glean", "Initialised android glean")
+    }
+    @JvmStatic
+    fun setGleanUploadEnabled(upload: Boolean) {
+        Glean.setUploadEnabled(upload)
+    }
+    @JvmStatic
+    fun setGleanSourceTag(tags: String = "") {
+        // val tag_list = tag.split(",")
+        // Glean.setSourceTags(tag_list);
+        // TODO: Glean-Android does not support this, yet.

```

Comment:
I filed a bug about it, currently this APi is only available for debugging in Android which I believe does not make sense. See https://bugzilla.mozilla.org/show_bug.cgi?id=1766457